File: RinhabackendGo/model/transaction.go
package model

import "time"

type Transaction struct {
	Id          *int64     `json:"id" db:"ID"`
	ClientId    int        `json:"cliente_id" db:"CLIENTE_ID"`
	Value       int        `json:"valor" db:"valor"`
	Type        string     `json:"tipo" db:"tipo"`
	Description string     `json:"descricao" db:"descricao"`
	CreatedAt   *time.Time `json:"realizada_em" db:"realizada_em"`
}

type TransactionBalance struct {
	Value       int        `json:"valor" db:"valor"`
	Type        string     `json:"tipo" db:"tipo"`
	Description string     `json:"descricao" db:"descricao"`
	CreatedAt   *time.Time `json:"realizada_em" db:"realizada_em"`
}

File: RinhaBackendSpringBoot/src/main/java/br/com/ronistone/rinhabackendspringboot/model/Transaction.java
package br.com.ronistone.rinhabackendspringboot.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.OffsetDateTime;

public class Transaction {

    @JsonProperty("id")
    public Integer id;

    @JsonProperty("cliente_id")
    public Integer clientId;

    @JsonProperty("valor")
    public Integer value;

    @JsonProperty("tipo")
    public String type;

    @JsonProperty("descricao")
    public String description;

    @JsonProperty("realizada_em")
    @DateTimeFormat(pattern="yyyy-MM-ddTHH:mm:ssZ")
    public OffsetDateTime createAt;

}
